geointel/image_utils.py

```
@@ -0,0 +1,65 @@
+            mime_type = response.headers.get('content-type', 'image/jpeg')
+            # Handle cases where content-type includes charset or other parameters
+            if ';' in mime_type:
+                mime_type = mime_type.split(';')[0].strip()
+            
+            # Fallback to image/jpeg if MIME type is not recognized
+            if not mime_type.startswith('image/'):
+                mime_type = 'image/jpeg'
+            
+            return base64.b64encode(response.content).decode('utf-8'), mime_type
+        except requests.exceptions.ConnectionError:
+            raise ValueError(f"Failed to connect to URL: {image_path}. Please check your internet connection.")
```

Ruff (0.14.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)